src/mam4xx/hetfrz.hpp

```diff
@@ -0,0 +1,1490 @@
+      (Ktherm_air + 2.5 * Kn * Ktherm) /
+      ((1.0 + 3.0 * Kn) * (2.0 * Ktherm_air + 5.0 * Kn * Ktherm + Ktherm));
+  const Real Q_heat = Ktherm_air / r3lx *
+                      (1.0 + 0.3 * haero::sqrt(Re) * haero::pow(Pr, 0.33)) *
+                      Tdiff;
+  const Real K_thermo_cotton =
+      4.0 * pi * haero::square(r3lx) * f_t * Q_heat / pressure;
+  const Real K_diffusio_cotton =
+      -(1.0 / f_t) * (r_gas_h2o_vapor * temperature / latvap) * K_thermo_cotton;
+
+  K_total = 1.e6 * (K_brownian + K_thermo_cotton +
+                    K_diffusio_cotton); // convert m3/s -> cm3/s
+  K_total = haero::max(0.0, K_total);
+}
+
+KOKKOS_INLINE_FUNCTION
+void collkernel(const Real temperature, const Real pressure, const Real eswtr,
+                const Real r3lx, const Real r_bc, const Real r_dust_a1,
+                const Real r_dust_a3, Real &Kcoll_bc, Real &Kcoll_dust_a1,
+                Real &Kcoll_dust_a3) {
+
+  // thermal conductivities from Seinfeld & Pandis, Table 8.6
+  constexpr Real Ktherm_bc = 4.2;    // Carbon
+  constexpr Real Ktherm_dust = 0.72; // Clay
+
+  const Real tc = temperature - Constants::freezing_pt_h2o;
+
+  // air viscosity for tc<0,
+  const Real viscos_air = get_air_viscosity(tc);
+
+  const Real r_gas_dry_air = 287.04231136504870; // (BAD CONSTANT)
+  // air density
+  const Real rho_air = pressure / (r_gas_dry_air * temperature);
+
+  // mean free path: Seinfeld & Pandis 8.6
+  const Real lambda =
+      2.0 * viscos_air /
+      (pressure *
+       haero::sqrt(8.0 / (Constants::pi * r_gas_dry_air * temperature)));
+
+  // latent heat of vaporization, varies with T
+  const Real latvap = get_latent_heat_vapor(tc);
+
+  // Reynolds number
+  const Real Re = get_reynolds_num(r3lx, rho_air, viscos_air);
+
+  // thermal conductivity of air: Seinfeld & Pandis eq. 15.75
+  const Real Ktherm_air = 1.0e-3 * (4.39 + 0.071 * temperature); // J/(m s K)
+
+  // Prandtl number
+  const Real Pr = viscos_air * Constants::cp_dry_air / Ktherm_air;
+
+  // calculate T-Tc as in Cotton et al.
+  const Real Tdiff =
+      get_temperature_diff(temperature, pressure, eswtr, latvap, Ktherm_air);
+
+  calculate_collkernel_sub(temperature, pressure, rho_air, r3lx, r_bc, lambda,
+                           latvap, viscos_air, Re, Ktherm_air, Ktherm_bc, Pr,
+                           Tdiff, Kcoll_bc);
+
+  calculate_collkernel_sub(temperature, pressure, rho_air, r3lx, r_dust_a1,
+                           lambda, latvap, viscos_air, Re, Ktherm_air,
+                           Ktherm_dust, Pr, Tdiff, Kcoll_dust_a1);
+
+  calculate_collkernel_sub(temperature, pressure, rho_air, r3lx, r_dust_a3,
+                           lambda, latvap, viscos_air, Re, Ktherm_air,
+                           Ktherm_dust, Pr, Tdiff, Kcoll_dust_a3);
+}
+
+KOKKOS_INLINE_FUNCTION
+Real get_form_factor(const Real alpha) {
+  const Real v_cos = cos(alpha); // This should resolve to haero cos
```

Comment:
You could sidestep this comment and just use `haero::cos` if you like.

Reply:
Ah, yes, I think I actually added `haero::cos` and then forgot to use it.